clarifying resource-sharing perspectives (#780)
opensciencegrid/docs: docs/resource-sharing/overview.md

Compute Resource Sharing Overview
=================================

OSG uses a resource-overlay ("pilot") model to share resources from your local cluster:

> compute resources are added to a large central resource pool in OSG through the use of a bootstrap process, often called a

Reviewer: ```suggestion
compute resources are added to a large central resource pool in the OSG through the use of a bootstrap process, often called a
```

_pilot_ or a _glidein_.

> These pilots, in turn, download and execute OSG user jobs (also known as "payloads") from the resource pool to run within the pilots.

Reviewer: We use OSG in previous sentences a lot so I think the additions of "OSG" here are unnecessary and possibly confusing since OSG is a pretty overloaded term (OSG consortium or OSG VO?)

Author: I think it's necessary. For someone who doesn't already understand how things work, it's very confusing. The reason I came into make suggestions was that I had to read over each option several times, myself, and try to insert my prior knowledge - which others will not have - to make sure that I understand.

Reviewer: I'm fine with keeping the first one (see suggestion above) but "OSG user jobs" is confusing for folks that know about the OSG VO, which is another large constituency that we have to serve.

Author: Can you elaborate on the confusion you're worried about? I'm not sure I follow.

Reviewer: OSG user jobs could be understood as "OSG VO user jobs", which makes the meaning of the sentence confusing for other VO in the OSG consortium.

Author: "user jobs from OSG"?

Reviewer: https://github.com/opensciencegrid/docs/pull/780/files#r577088576 is the happy compromise that we agreed upon

> These pilots, in turn, download and execute OSG user jobs (also known as "payloads") from the resource pool to run within the pilots.

Reviewer: ```suggestion
These pilots, in turn, download and execute user jobs from an OSG community (also known as "payloads") from the resource
pool to run within the pilots.
```

Suggestion to apply what we discussed @lmichael107

There are several ways pilots can join a resource pool:

> * Submitted to your local batch system by a [*compute entrypoint*](../compute-element/htcondor-ce-overview.md).
>   These jobs are created by an external entity, a *pilot factory* based on observed demand in the pool. The
>   *compute entrypoint* (CE) is the most common way pilots are created as it is highly automated and responsive
>   to existing demand.

Reviewer: ```suggestion
* Submitted to your local batch system by a [*compute entrypoint*](../compute-element/htcondor-ce-overview.md) (CE).
  These jobs are created by an external entity, a *pilot factory* based on observed demand in the pool.
  The CE is the most common way to receive pilot jobs since they integrate with automated processes that are responsive
  to existing demand.
```

> * Sites can launch [*pilot containers*](os-backfill-containers.md) when they have local resources they
>   would like to contribute directly to a specific OSG pool. The site-launched *pilot container* method is useful
>   for backfilling resources expected to be persistently idle; however, at times these pilots may stay idle
>   because there is insufficient demand within the resource pool.

Reviewer: ```suggestion
* Sites can launch [*pilot containers*](os-backfill-containers.md) when they have local resources they
  would like to contribute directly to a specific OSG pool.
  The site-launched *pilot container* method is useful for backfilling resources without the need for a batch system;
  however, at times these pilots may stay idle because there is insufficient demand within the resource pool.
```

We aim to recommend container deployments that can backfill in a similar way to preemptible Slurm backfill partitions

* Users on your local cluster can launch *personal pilot containers* within the local batch system so that these
  receive their OSG user jobs.

> 
> 
> 

Reviewer: ```suggestion
```

Just dropping extra whitespace

There are several ways pilots can join a resource pool:

* Submitted to your local batch system by a [*compute entrypoint*](../compute-element/htcondor-ce-overview.md) (CE).
  These jobs are created by an external entity, a *pilot factory* based on observed demand in the pool.
  The CE is the most common way to receive pilot jobs since they integrate with automated processes that are responsive
  to existing demand.
* Sites can launch [*pilot containers*](os-backfill-containers.md) when they have local resources they
  would like to contribute directly to a specific OSG pool.
  The site-launched *pilot container* method is useful for backfilling resources without the need for a batch system;
  however, at times these pilots may stay idle because there is insufficient demand within the resource pool.

> * Users on your local cluster can launch *personal pilot containers* within the local batch system so that these
>   receive their OSG user jobs.

Reviewer: ```suggestion
* Users can launch *personal pilot containers* within a site's batch system so they can use an existing share or
  allocation at a site through the open science pool.
```

Oh I missed this change in previous reviews. I prefer the old wording since this sounds like the personal pilot containers (different from backfill containers) do not just accept "OSG user jobs". They'd only accept a specific user's jobs -- the person that started up the container: https://opensciencegrid.org/docs/resource-sharing/user-containers/

Reviewer: Hi @lmichael107 I'm reverting your wording for this bullet here to get the rest of the PR merged in. If you'd like to continue working on the wording of this bullet, please submit another pull request
